docs/puzzles/day6.md

```diff
@@ -2,12 +2,336 @@
+map of occurrences? We create a new map of occurrences where we associate to 
+the timer value `0` the number of current occurrences for the timer value `1`, 
+we associate to the timer value `1` the number of current occurrences for 
+the timer value `2`, and so on, to model time passing. However, there are 
+two special cases due to the fact that every seven days a fish creates 
+another fish. The number of fish whose timer value is `6` is the current 
+number of fish whose timer value is `7` _plus_ the number of fish whose 
+timer value is `0` (those fish created a new fish, and they will create 
+another fish in `6` days). Also, the number of fish whose timer value is `8` 
+is the number of fish that have been created during this turn, that is the 
+current number of fish whose timer value is `0`. This leaves us with the 
+following implementation:
+
+~~~ scala
+def tick(population: Map[Int, BigInt]): Map[Int, BigInt] =
+  def countPopulation(daysLeft: Int): BigInt = population.getOrElse(daysLeft, BigInt(0))
+  Map(
+    0 -> countPopulation(1),
+    1 -> countPopulation(2),
+    2 -> countPopulation(3),
+    3 -> countPopulation(4),
+    4 -> countPopulation(5),
+    5 -> countPopulation(6),
+    6 -> (countPopulation(7) + countPopulation(0)),
+    7 -> countPopulation(8),
+    8 -> countPopulation(0)
+  )
+~~~
+
+The last missing piece to answer the challenge is to run the simulation for 
+a given number of days, and to compute the number of fish at the end of the 
+simulation:
+
+~~~ scala
+def simulate(days: Int, initialPopulation: Map[Int, BigInt]): BigInt =
+  (1 to days)
+    .foldLeft(initialPopulation)((population, _) => tick(population))
+    .values
+    .sum
+~~~
+
+The method `simulate` is very similar to the one I wrote for part 1. The 
+only difference is how it computes the number of fish from the model. It 
+achieves this by summing the groups of fish: the method `values` returns a 
+collection of groups of fish (each containing the number of fish in that 
+group), finally the method `sum` sums up the groups.
+
+## Final code for part 2
+
+~~~ scala
+// "How many lanternfish would there be after 256 days?"
+def part2(input: String): BigInt =
+  simulate(
+    days = 256,
+    Fish.parseSeveral(input).groupMapReduce(_.timer)(_ => BigInt(1))(_ + _)
+  )
+
+def simulate(days: Int, initialPopulation: Map[Int, BigInt]): BigInt =
+  (1 to days)
+    .foldLeft(initialPopulation)((population, _) => tick(population))
+    .values
+    .sum
+
+def tick(population: Map[Int, BigInt]): Map[Int, BigInt] =
+  def countPopulation(daysLeft: Int): BigInt = population.getOrElse(daysLeft, BigInt(0))
+  Map(
+    0 -> countPopulation(1),
+    1 -> countPopulation(2),
+    2 -> countPopulation(3),
+    3 -> countPopulation(4),
+    4 -> countPopulation(5),
+    5 -> countPopulation(6),
+    6 -> (countPopulation(7) + countPopulation(0)),
+    7 -> countPopulation(8),
+    8 -> countPopulation(0)
+  )
+~~~
+
+<Solver puzzle="day6-part2"/>
```

Comment:
The import is missing: 
```suggestion
import Solver from "../../../../website/src/components/Solver.js"
<Solver puzzle="day6-part2"/>
```

Reply:
Actually it should be placed before `<Solver puzzle="day6-part1"/>` (instead of part2). The beginning of the file, after the `---` may be the best place to put it.

Reply:
Fixed in 130ce4e